helix-term/src/ui/picker.rs

```diff
@@ -18,35 +18,281 @@ use std::{
     borrow::Cow,
     cmp::Reverse,
     collections::HashMap,
+    fs,
     io::Read,
     path::{Path, PathBuf},
 };
 
-use crate::ui::{Prompt, PromptEvent};
+use crate::ui::{overlay, Prompt, PromptEvent};
 use helix_core::{movement::Direction, Position};
 use helix_view::{
     editor::Action,
     graphics::{CursorKind, Margin, Modifier, Rect},
     Document, Editor,
 };
 
-pub const MIN_AREA_WIDTH_FOR_PREVIEW: u16 = 72;
-/// Biggest file size to preview in bytes
-pub const MAX_FILE_SIZE_FOR_PREVIEW: u64 = 10 * 1024 * 1024;
+// based on exa but not sure where to put this
+/// More readable aliases for the permission bits exposed by libc.
+#[cfg(unix)]
+mod modes {
+    // The `libc::mode_t` type’s actual type varies, but the value returned
+    // from `metadata.permissions().mode()` is always `u32`.
+    pub type Mode = u32;
+
+    pub const USER_READ: Mode = libc::S_IRUSR as Mode;
+    pub const USER_WRITE: Mode = libc::S_IWUSR as Mode;
+    pub const USER_EXECUTE: Mode = libc::S_IXUSR as Mode;
+
+    pub const GROUP_READ: Mode = libc::S_IRGRP as Mode;
+    pub const GROUP_WRITE: Mode = libc::S_IWGRP as Mode;
+    pub const GROUP_EXECUTE: Mode = libc::S_IXGRP as Mode;
+
+    pub const OTHER_READ: Mode = libc::S_IROTH as Mode;
+    pub const OTHER_WRITE: Mode = libc::S_IWOTH as Mode;
+    pub const OTHER_EXECUTE: Mode = libc::S_IXOTH as Mode;
+
+    pub const STICKY: Mode = libc::S_ISVTX as Mode;
+    pub const SETGID: Mode = libc::S_ISGID as Mode;
+    pub const SETUID: Mode = libc::S_ISUID as Mode;
+}
 
-/// File path and range of lines (used to align and highlight lines)
-pub type FileLocation = (PathBuf, Option<(usize, usize)>);
+#[cfg(not(unix))]
+mod modes {}
+
+// based on exa but not sure where to put this
+mod fields {
+    use super::modes;
+    use std::fmt;
+    use std::fs::Metadata;
+    #[cfg(unix)]
+    use std::os::unix::fs::{FileTypeExt, MetadataExt};
```

Comment:
I am not sure how do I fix this conditional compilation and get it working on windows.

Reply:
Oh, seemed like it was fixed.